edb/pgsql/delta.py

```diff
@@ -3263,19 +3265,57 @@ def create_index(cls, index, schema, context):
         else:
             except_src = None
 
+        sql_kwarg_exprs = dict()
+        # Get the name of the root index that this index implements
+        orig_name = sn.shortname_from_fullname(index.get_name(schema))
+        if orig_name == s_indexes.DEFAULT_INDEX:
+            root_name = orig_name
+        else:
+            root = index.get_root(schema)
+            root_name = root.get_name(schema)
+
+            kwargs = index.get_concrete_kwargs(schema)
+            # Get all the concrete kwargs compiled (they are expected to be
+            # constants)
+            # These are expected to be constants, so we don't have anchors,
+            # path prefixes, etc.
+            kw_options = qlcompiler.CompilerOptions(
+                modaliases=context.modaliases,
+                schema_object_context=cls.get_schema_metaclass(),
+                apply_query_rewrites=False,
+            )
+            for name, expr in kwargs.items():
+                # XXX: origin messes up compilation, but by this point we
+                # shouldn't care about the expression's origin.
+                expr.origin = None
+                kw_expr = expr.ensure_compiled(
+                    schema=schema,
+                    options=kw_options,
+                    as_fragment=True,
+                )
+                kw_ir = kw_expr.irast
+                kw_sql_tree = compiler.compile_ir_to_sql_tree(
+                    kw_ir.expr, singleton_mode=True)
+                sql = codegen.SQLSourceGenerator.to_source(kw_sql_tree)
+                # HACK: the compiled SQL is expected to have some unnecessary
```

Comment:
Hmmmmm this is very sketchy. What casts were running afoul of this?

Reply:
+1.